project/src/components/main/main.tsx

```diff
@@ -1,15 +1,28 @@
 import Locations from '../locations/locations';
 import Map from '../map/map';
 import Header from '../header/header';
-import {Offers} from '../../mocks/offer';
+import {Offer, Offers} from '../../mocks/offer';
 import OffersList from '../offers-list/offers-list';
+import {useState} from 'react';
+import {MAP_HEIGHT} from '../../consts';
 
 
 type MainProps = {
   offers: Offers;
 }
 
 function Main({offers}: MainProps): JSX.Element {
+  const city = offers[0].city;
+
+  const [activeCard, setActiveCard] = useState< Offer | undefined>(
+    undefined,
+  );
+
+  const onListItemHover = (id: string) => {
+    const currentOffer = offers.find((offer) => String(offer.id) === id);
+    setActiveCard(currentOffer);
+  };
```

Comment:
```suggestion
  const [activeCard, setActiveCard] = useState< Offer | null>(
    null,
  );

  const onListItemHover = (id: string) => {
    const currentOffer = offers.find((offer) => String(offer.id) === id);
    setActiveCard(currentOffer ?? null);
  };
```
лучше так. не используй undefined